op-challenger/game/fault/types/types.go

```diff
@@ -105,6 +105,12 @@ type TraceProvider interface {
 
 	// AbsolutePreState is the pre-image value of the trace that transitions to the trace value at index 0
 	AbsolutePreState(ctx context.Context) (preimage []byte, err error)
+
+	// AbsolutePreStateCommitment is the commitment of the pre-image value of the trace that transitions to the trace value at index 0
+	AbsolutePreStateCommitment(ctx context.Context) (hash common.Hash, err error)
+
+	// StateHash computes the state-hash of the given state, or returns an error if the state is invalid.
+	StateHash(ctx context.Context, state []byte) (common.Hash, error)
```

Comment:
This function is included in the `TraceProvider` due to the fact that there may be different post-processing tasks (i.e. the VM status byte) for different `TraceProviders` in the future. The thought was that having each `TraceProvider` define how a given state should be hashed + [pre/post]processed will likely save us some headaches in the future when there are more games.

Reply:
It may be useful in the future but right now it's largely leaking an internal detail and not adding any value.  I'm not strongly opinionated but generally I'd apply YAGNI (you ain't gonna need it) and only add it when we actually have a use for it.